### DISCUSSION TITLE
Getting logging into VSCode

### DISCUSSION THREAD

**Discussion by xldenis** [c-cube/linol#28] [General]
Do you have any idea how I could set up a reporter which could log into an output channel in VSCode? So far I've been using a manual wrapper around `notify_back#send_log_msg` but I would like to be able to just use the regular `Logs` API. 

Am I missing something? Or is there a different approach that servers generally take to store their logs?

**Comment by c-cube** [comment]
I think I log to /tmp typically, but this could fit the bill perhaps?

https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#logTrace

**Reply by xldenis** [reply]
Yes, but I wanted to hook into the `Logs` interface so that the mesages would be sent appropriately. I think I might be able to hack something together so that a reporter is installed once the server starts up but it's a kind of impedance mismatch. 

The `"verbose"` setting causes a client to _display_ the LSP log messages (`send_log_msg`) not any other logs / stderr the server might be displaying.

**Reply by xldenis** [reply]
Sorry, I should clarify a bit what my actual problem is: I would like to not have to thread the `notify_back` object everywhere to send logs to the LSP client if there were some way for the server to provide the appropriate reporter that would be amazing, otherwise I suppose I'll make do with `/tmp` or something similar.

**Comment by c-cube** [comment]
I'm on my phone but I think we had prior discussions about being able to send server notifications/queries directly (notify_back is in the context of a single client query iirc). So that's the critical feature. 

Assuming we have that, and it's not very hard to do I think, we can just make a Logs reporter out of it as soon as the connection is established.

**Reply by xldenis** [reply]
Yea, I think that would be the perfect solution.

**Reply by c-cube** [reply]
see #29, not tested

**Reply by xldenis** [reply]
I was able to use #29 to implement a `Logs` reporter which properly sends log messages to the LSP client, its a pretty straightfoward adaptation of the `Logs_lwt` formatter.
